include/unifex/connect_awaitable.hpp

```diff
@@ -99,12 +114,27 @@ class _sender_task<Receiver>::type {
         friend auto tag_invoke(CPO cpo, const promise_type& p) noexcept(
             std::is_nothrow_invocable_v<CPO, const Receiver&>)
             -> std::invoke_result_t<CPO, const Receiver&> {
-      return cpo(std::as_const(p.receiver_));
+      if constexpr (
+          WithAsyncStackSupport && same_as<CPO, tag_t<get_async_stack_frame>>) {
+        return &p.frame_;
+      } else {
+        return std::move(cpo)(std::as_const(p.receiver_));
+      }
```

Comment:
just a nit and not a big deal. this is a weird check. is it possible to separately define a get_async_stack_frame tag_invoke that will be called instead of this generic CPO function?

Reply:
It is weird, but I ran into trouble (that I don't remember in detail now) expressing the conditions correctly on a separate `tag_invoke` overload.